app/pages/sections/commencement-date/getDateErrors.js

```diff
@@ -0,0 +1,44 @@
+export const getDateErrors = (data) => {
+  const errorsMap = [
+    {
+      field: 'CommencementDate',
+      part: ['day', 'month', 'year'],
+      id: 'CommencementDateRequired',
+    },
+    {
+      field: 'CommencementDate',
+      part: ['day'],
+      id: 'CommencementDateDayRequired',
+    },
+    {
+      field: 'CommencementDate',
+      part: ['month'],
+      id: 'CommencementDateMonthRequired',
+    },
+    {
+      field: 'CommencementDate',
+      part: ['year'],
+      id: 'CommencementDateYearRequired',
+    },
+    {
+      field: 'CommencementDate',
+      part: ['day', 'month'],
+      id: 'CommencementDateNotReal',
+    },
+  ];
```

Comment:
rather that this being an array. how about making this an object. Using the id as the key also.  That way when we do the lookup below we can use the id rather than the index value.

Reply:
okay dokey

Reply:
but the reason I did it like this is that it preserves the format that is needed for the errors to be processed later
so you want e.g. {
    CommencementDateDayRequired: {
      field: 'CommencementDate',
      part: ['day'],
      id: 'CommencementDateDayRequired',
    },
}

Reply:
yes something like that is perfect. I don't mind the id being duplicated but just to make the look up look cleaner and not use index values